plugins/modem-manager/fu-mm-device.c

```diff
@@ -296,7 +346,7 @@ fu_mm_device_probe_default (FuDevice *device, GError **error)
 	fu_device_set_version (device, version);
 	for (guint i = 0; device_ids[i] != NULL; i++)
 		fu_device_add_instance_id (device, device_ids[i]);
-	if (fu_device_get_vendor_ids (device) == NULL) {
+	if (fu_device_get_vendor_ids (device) != NULL) {
```

Comment:
This seems like a wrong logic change, can you please elaborate?

Reply:
In the process of get-devices, fu_device_get_vendor_ids (device) will return a GPtrArray*(not NULL), if use ==, code can't enter the block to set vendor ID, the result is that the device display no vendor id and can not upgrade.

Reply:
Ah so it's an existing logic error from when multiple vendor_ids support was added.  I guess then what you want to be doing is change the check to something like

```
vendors = fu_device_get_vendor_ids (device);
if (vendors == NULL || vendors->len == 0)
```

Reply:
Yes, I think your code logic is better, thanks.